herbert_core/utilities/classes/@IX_dataset_2d/private/plot_twod.m

```diff
@@ -1,247 +1,289 @@
+    ind = stringmatchi(plot_type, plot_types);
     if ~isempty(ind)
-        plot_type=plot_types{ind};
+        plot_type = plot_types{ind};
     else
-        mess='Plot type not recognised';
-        if nargout<=3, error(mess), end
+        error('HERBERT:graphics:invalid_argument', ...
+            'Plot type ''%s'' is not recognised', plot_type);
     end
 else
-    mess='Plot type must be a (non-empty) character string';
-    if nargout<=3, error(mess), end
+    error('HERBERT:graphics:invalid_argument', ...
+        'Plot type must be a non-empty character string. It is ''%s''', ...
+        disp2str(plot_type));
 end
 
-% Get figure name or figure handle - used to branch later on
-if strcmpi(plot_type,'area')        % area plot
-    default_fig_name=get_global_var('genieplot','name_area');
-elseif strcmpi(plot_type,'surface') % surface plot
-    default_fig_name=get_global_var('genieplot','name_surface');
-elseif strcmpi(plot_type,'surface2')% surface2 plot
-    default_fig_name=get_global_var('genieplot','name_surface');
-elseif strcmpi(plot_type,'contour') % contour plot
-    default_fig_name=get_global_var('genieplot','name_contour');
-else
-    error('Logic error: unrecognised plot_type')
-end
-[fig_out,ok,mess]=genie_figure_target(fig,newplot,default_fig_name);
-if ~ok
-    error(mess);
-end
-
-% Check plot limits:
-if ~newplot
-    if ~isempty(par)
-        mess='Cannot specify plot limits when overplotting';
-        if nargout<=3, error(mess), end
-    end
-    % Need to specify xlims,ylims not give in case need to create a figure
-    xlims=false;
-    ylims=false;
-    zlims=false;
-else
-    if isempty(par)
-        xlims=false;
-        ylims=false;
-        zlims=false;
-    elseif numel(par)==2||numel(par)==4||numel(par)==6
-        bad=false;
-        xlims=true;
-        if isnumeric(par{1}) && isscalar(par{1}), xlo=par{1}; else bad=true; end
-        if isnumeric(par{2}) && isscalar(par{2}), xhi=par{2}; else bad=true; end
-        if numel(par)>=4
-            ylims=true;
-            if isnumeric(par{3}) && isscalar(par{3}), ylo=par{3}; else bad=true; end
-            if isnumeric(par{4}) && isscalar(par{4}), yhi=par{4}; else bad=true; end
-        else
-            ylims=false;
+% Set the default figure name for the requested plot_type and configuration
+% parameters for argument parsing
+switch plot_type
+    case 'area'
+        if ~is_string(default_fig_name)
+            default_fig_name = 'Herbert area plot';
         end
-        if numel(par)>=6
-            zlims=true;
-            if isnumeric(par{5}) && isscalar(par{5}), zlo=par{5}; else bad=true; end
-            if isnumeric(par{6}) && isscalar(par{6}), zhi=par{6}; else bad=true; end
-        else
-            zlims=false;
+        alternate_cdata_ok = false;
+        w1_data_name = '';
+        w2_data_name = '';
+    case 'surface'
+        if ~is_string(default_fig_name)
+            default_fig_name = 'Herbert surface plot';
```

Comment:
checking if property undefined using invalid property type is a bit convoluted.  Do you really want to distinguish between setting figure name to '' or [] at input? 
Ok -- I see that you are really do... Looks like overcomplication.

Reply:
Matlab allows figures with no name - when the corresponding figure handle property `Name` is set to ''. Therefore the undefined status has to be set to something different than ''. It was something that I found a bit irritating, for the same reason as you find it baroque, but when I thought it through at the time there were circumstances I could see that caused problems if I didn't distinguish between the two cases. For that reason I settled on unassigned as [] (in just the same way as Matlab sets cell array elements to [] when you do something like myvar = cell(1,5) ).